pkg/services/ngalert/state/manager.go

```diff
@@ -212,6 +212,7 @@ func (st *Manager) maybeTakeScreenshot(
 
 // Set the current state based on evaluation results
 func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRule, result eval.Result, extraLabels data.Labels) *State {
+	logger := st.log.New(alertRule.GetKey().LogContext()...)
```

Comment:
I wonder if we can just accept logger as parameter. The pretty much same is implemented in ProcessEvalResults that calls it